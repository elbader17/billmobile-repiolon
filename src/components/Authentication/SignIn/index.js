import React from 'react';
import { StyleSheet, View } from 'react-native';
import { Input, Button } from "react-native-elements";
import { signIn } from '../../../app/authentication/actions';

class SignIn extends React.Component {

    constructor(props) {
      super(props);
      this.state = {
        email: '',
        password: '@Am1234!',
        name:'',
        confirmPassword: '@Am1234!',
      };
    }

    handleSignIn = () => {
      const { email, password } = this.state;
      const { dispatch } = this.props;
      dispatch(signIn(email, password));
    }

    setEmail = (value) => this.setState({ email: value })
    setPassword = (value) => this.setState({ password: value })

    render() {
      return(
          <View>
            <Input
              label="Email"
              leftIcon={{ type: 'font-awesome', name: 'envelope' }}
              onChangeText={this.setEmail}
              placeholder="my@email.com"
            />
            <Input
              label="Password"
              leftIcon={{ type: 'font-awesome', name: 'lock' }}
              onChangeText={this.setPassword}
              placeholder="p@ssw0rd123"
              secureTextEntry
            />
            <Button
              title='Submit'
              onPress={ this.handleSignIn }
            />
        </View>
      )
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
});


export default SignIn;
