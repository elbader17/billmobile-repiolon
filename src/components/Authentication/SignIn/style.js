import { StyleSheet } from 'react-native';
import { COLORS } from '../../../constants/colors';
import { FONTS } from '../../../constants/fonts';
import { METRICS } from '../../../constants/metrics';

const styles = StyleSheet.create({
  container:{
    width: METRICS.screenWidth,
    height: METRICS.screenHeight
  },
  container2:{
    alignItems: 'center',
    paddingHorizontal: 45,
    paddingTop: 15
  },
  textBoxBtnHolder: {
    position: 'relative',
    alignSelf: 'stretch',
    justifyContent: 'center',
    marginTop: 5
  },
  textBox: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size18,
    alignSelf: 'stretch',
    height: 45,
    paddingRight: 45,
    paddingLeft: 8,
    borderWidth: 1,
    paddingVertical: 0,
    borderColor: COLORS.gray,
    borderRadius: 2
  },
  visibilityBtn: {
    position: 'absolute',
    right: 3,
    height: 40,
    width: 35,
    padding: 5
  },
  btnImage: {
    resizeMode: 'contain',
    height: '100%',
    width: '100%',
    opacity: 0.4
  },
  submit: {
    backgroundColor: COLORS.red,
    width: 266,
    height: 40,
    borderRadius: 2
  },
  submitDisabled: {
    backgroundColor: COLORS.redLight,
    width: 266,
    height: 40,
    borderRadius: 2
  },
  submitText: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size14,
    color: COLORS.white
  },
  textRegister: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size11,
    color: COLORS.grayDark,
    paddingVertical: 15
  },
  red: {
    color: COLORS.red
  },
  textFooterA: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size11,
    color: COLORS.grayDark,
    marginTop: 110
  },
  textFooterB: {
    fontFamily: FONTS.latoRegular,
    fontSize: FONTS.size11,
    color: COLORS.grayDark,
    fontWeight: 'bold'
  }
});

export default styles;