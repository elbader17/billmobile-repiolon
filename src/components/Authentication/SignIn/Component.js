import React from 'react';
import { View, Text, TextInput, Image, TouchableOpacity } from 'react-native';
import { showMessage } from "react-native-flash-message";
import LinearGradient from 'react-native-linear-gradient';
import { 
  messageErrorSignIn,
  messageErrorSignInConfirm,
  messageErrorSignInExist,
  messageErrorSignInData
} from '../../../utils/messagesNotifications';
import { Button } from "react-native-elements";
import style from '../SignUp/style';
import { GRADIENTYELLOW } from '../../../constants/colors';


const EMAIL_REGEXP = /^[a-zA-Z0-9]+@[a-zA-Z0-9]+\.[A-Za-z]+$/;
const PASSWORD_REGEXP = new RegExp("^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})");

class SignIn extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      email: '@mozej.com',
      password: '@Martin44',
      hidePassword: true,
      loading: false
    };
  }

  managePasswordVisibility = () => {
    this.setState({ hidePassword: !this.state.hidePassword });
  }

  handleSignIn = () => {
    const { email, password } = this.state;
    const { signIn } = this.props;
    this.setLoading(true);
    signIn(email, password)
    .then( (_data) => {
      if(this.props.jwtToken !== ''){
        console.log(this.props.jwtToken);
        this.navigateNext();
      } else {
        if ( _data.message === 'User does not exist.'){
          showMessage(messageErrorSignInExist)
          this.props.navigation.navigate('Authentication');
        }else if ( _data.message === 'User is not confirmed.'){
          showMessage(messageErrorSignInConfirm);
          this.props.navigation.navigate('ConfirmationCodeRegister', { email: this.state.email });
        }else {
          showMessage(messageErrorSignInData);
          this.props.navigation.navigate('Authentication');
        }
        this.setLoading(false);
      }
    })
    .catch(err => showMessage(messageErrorSignIn)); //incluir err.message
  }

  navigateNext = () => {
    if (this.props.fiscalIdentityComplete) {
      this.props.navigation.navigate('Home');
    } else{
      this.props.navigation.navigate('Configure');
    }
  }

  validateData = () => {
    const isValidPassword = PASSWORD_REGEXP.test(this.state.password);
    const isValidEmail = EMAIL_REGEXP.test(String(this.state.email).toLowerCase());
    return (isValidPassword && isValidEmail);
  }

  setEmail = (value) => this.setState({ email: value })
  setPassword = (value) => this.setState({ password: value })
  setLoading = (bool) => this.setState({ loading: bool })

  render() {
    const hide = require('../../../images/hide.png')
    const show = require('../../../images/show.png')
    return(
      <View style={style.container}>
        <View style={style.containerInputs}>
          <View style={ style.textBoxBtnHolder }>
            <TextInput
              label="Email"
              onChangeText={this.setEmail}
              value={this.state.email}
              placeholder="Tu email"
              style={ style.textBox }
              keyboardType='email-address'
            />
          </View>
          <View style={ style.textBoxBtnHolder }>
            <TextInput
              label="Password"
              onChangeText={this.setPassword}
              value={this.state.password}
              placeholder="ContraseÃ±a"
              style={ style.textBox }
              secureTextEntry={ this.state.hidePassword }
            />
            <TouchableOpacity
              activeOpacity={ 0.8 }
              style={ style.visibilityBtn }
              onPress={ this.managePasswordVisibility }
            >
              <Image
                source={( this.state.hidePassword ) ? hide : show }
                style={ style.btnImage }
              />
            </TouchableOpacity>
          </View>
          <LinearGradient
            colors={ GRADIENTYELLOW }
            start={{x: 0.0, y: 1.0}} 
            end={{x: 1.0, y: 1.0}}
            style={{borderRadius: 25, marginTop: 15}}
          >
            <Button
              title='ENTRAR'
              testID={'submitSignIn'}
              onPress={ this.handleSignIn }
              buttonStyle={ style.submit }
              titleStyle={ style.textRegular14WhiteBold }
              disabledTitleStyle={ style.textRegular14WhiteBold}
              disabledStyle={ style.submitDisabled }
              disabled={ !this.validateData() }
              loading = {this.state.loading}
            />
          </LinearGradient>
        </View>
        
      </View>
    )
  }
}

export default SignIn;
