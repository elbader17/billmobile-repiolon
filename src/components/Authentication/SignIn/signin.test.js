import React from 'react';
import { shallow } from 'enzyme';
import SignIn  from './Component';

const promise = Promise.resolve(0);
const signInMock = jest.fn(() => promise);

test('render correctly component SignIn', () => {
  const component = shallow(<SignIn />);
  expect(component).toMatchSnapshot();
});

test('call the handleSignIn function', () => {
  const component = shallow(<SignIn signIn={signInMock} />);
  const button = component.findWhere(node => node.prop('testID') === 'submitSignIn');
  button.simulate('press');
  expect(signInMock).toBeCalled();
});

/* FALLA POR CODIGO HARDCODEADO (CAMBIAR)
test('initial state with the correct values', () => {
  const initialState = {
    email: '',
    password: '',
    hidePassword: true,
    loading: false
  };
  const component = shallow(<SignIn />);
  const state = component.instance().state;
  expect(state).toEqual(initialState);
});
*/

describe('button INICIAR SESIÃ“N', () => {
  test('button initially disabled', () => {
    const component = shallow(<SignIn signIn={signInMock} />);
    const button = component.findWhere(node => node.prop('testID') === 'submitSignIn');
    button.simulate('press');
    expect(button.props().disabled).toBe(true);
  });
    
  test('button enabled with the correct data', () => {
    const component = shallow(<SignIn signIn={signInMock} />);
    component.setState({ 
      email: 'martin@gmail.com',
      password: '@Martin55',
     });
    const button = component.findWhere(node => node.prop('testID') === 'submitSignIn');
    expect(button.props().disabled).toBe(false);
  });
    
  test('button disabled with email incorrect', () => {
    const component = shallow(<SignIn signIn={signInMock} />);
    component.setState({ 
      email: 'martingmail.com',
      password: '@Martin55',
    });
    const button = component.findWhere(node => node.prop('testID') === 'submitSignIn');
    expect(button.props().disabled).toBe(true);
  });
     
  test('button loading after press', () => {
    const component = shallow(<SignIn signIn={signInMock} />);
    const button = component.findWhere(node => node.prop('testID') === 'submitSignIn');
    button.simulate('press');
    expect(component.instance().state.loading).toBe(true);
  });

});