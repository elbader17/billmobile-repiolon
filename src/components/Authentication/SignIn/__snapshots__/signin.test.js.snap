// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly component SignIn 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 30,
      "paddingVertical": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
          "marginTop": 5,
          "position": "relative",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        keyboardType="email-address"
        label="Email"
        onChangeText={[Function]}
        placeholder="Tu email"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "stretch",
            "borderColor": "#C4C4C4",
            "borderRadius": 2,
            "borderWidth": 1,
            "color": "#858585",
            "fontFamily": "Lato-Regular",
            "fontSize": 18,
            "height": 45,
            "paddingLeft": 8,
            "paddingRight": 45,
            "paddingVertical": 0,
          }
        }
        underlineColorAndroid="transparent"
        value="@mozej.com"
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
          "marginTop": 5,
          "position": "relative",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        label="Password"
        onChangeText={[Function]}
        placeholder="Contraseña"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "alignSelf": "stretch",
            "borderColor": "#C4C4C4",
            "borderRadius": 2,
            "borderWidth": 1,
            "color": "#858585",
            "fontFamily": "Lato-Regular",
            "fontSize": 18,
            "height": 45,
            "paddingLeft": 8,
            "paddingRight": 45,
            "paddingVertical": 0,
          }
        }
        underlineColorAndroid="transparent"
        value="@Martin44"
      />
      <TouchableOpacity
        activeOpacity={0.8}
        onPress={[Function]}
        style={
          Object {
            "height": 40,
            "marginTop": 2,
            "padding": 6,
            "position": "absolute",
            "right": 3,
            "width": 35,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/hide.png",
            }
          }
          style={
            Object {
              "height": "100%",
              "opacity": 0.4,
              "resizeMode": "contain",
              "width": "100%",
            }
          }
        />
      </TouchableOpacity>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#858585",
            "fontFamily": "Lato-Regular",
            "fontSize": 11,
          },
          Object {
            "paddingVertical": 15,
          },
        ]
      }
    >
      ¿No tienes una cuenta?
      <Text
        style={
          Object {
            "color": "#EE6123",
          }
        }
      >
         Registrate
      </Text>
    </Text>
    <ForwardRef(Themed.Button)
      buttonStyle={
        Object {
          "backgroundColor": "#EE6123",
          "borderRadius": 2,
          "height": 93.5,
          "width": 600,
        }
      }
      disabled={true}
      disabledStyle={
        Object {
          "backgroundColor": "#C4C4C4",
          "borderRadius": 2,
          "height": 93.5,
          "width": 600,
        }
      }
      disabledTitleStyle={
        Object {
          "color": "#fff",
          "fontFamily": "Lato-Semibold",
          "fontSize": 14,
        }
      }
      loading={false}
      onPress={[Function]}
      testID="submitSignIn"
      title="ENTRAR"
      titleStyle={
        Object {
          "color": "#fff",
          "fontFamily": "Lato-Semibold",
          "fontSize": 14,
        }
      }
    />
  </View>
</View>
`;
